pkg/metrics/metrics.go

```diff
@@ -69,14 +80,36 @@ func NewPrometheusNodeConfig(clientset *kubernetes.Clientset) (*PrometheusNodeCo
 	}, err
 }
 
+// newMetricsConfig creates a new instance for metricsConfig  to be used by the caller.
+func newMetricsConfig(cfg *rest.Config) (*metricsConfig, error) {
+	if cfg == nil {
+		return nil, errors.New("could not get config for creating metrics configuration")
+	}
+	oclient, err := k8sclient.NewForConfig(cfg)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not create config clientset")
+	}
+	mclient, err := monclient.NewForConfig(cfg)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not create monitoring clientset")
+	}
+	return &metricsConfig{oclient, mclient}, nil
+}
+
 // Validate will verify if cluster monitoring is enabled in the operator namespace
+// and delete any stale metrics resources from previous operator version
 func Validate(ctx context.Context, cfg *rest.Config, namespace string) error {
```

Comment:
This should be part of the `metricsConfig` struct. Please place this along with `removeStaleResources`.